src/main/java/org/skife/jdbi/v2/BeanMapper.java

```diff
@@ -51,6 +67,58 @@ public BeanMapper(Class<T> type)
         }
     }
 
+    /**
+     * Maps a property name to possible column names candidates. This default implementation will map the property name
+     * as-is, and converted from the camel-case syntax to an underscore separated name.
+     * <p>
+     * Processing of the returned mappings will be case insensitive. The mappings are not expected to change across
+     * invocations.
+     *
+     * @param propertyName the bean property name
+     *
+     * @return a collection holding the possible column names (mutable in the default implementation)
+     */
+    protected Collection<String> mapToColumnNames(String propertyName)
+    {
+        Locale locale = getLocale();
+
+        List<String> columnName = new ArrayList<String>(5);
+
+        // Add the bean property name as-is.
+        columnName.add(propertyName);
+
+        // Convert the property name from camel-case to underscores syntax. Freely adapted from Spring
+        // BeanPropertyRowMapper.
+        StringBuilder propertyNameWithUnderscores = new StringBuilder();
+        propertyNameWithUnderscores.append(propertyName.substring(0, 1));
+        for (int i = 1; i < propertyName.length(); i++) {
+            // Do case comparison using strings rather than chars (avoid to deal with non-BMP char handling).
+            String s = propertyName.substring(i, i + 1);
+            String slc = s.toLowerCase(locale);
+            if (!s.equals(slc)) {
+                // Different cases: tokenize.
+                propertyNameWithUnderscores.append("_").append(slc);
+            }
+            else {
+                propertyNameWithUnderscores.append(s);
+            }
+        }
+        columnName.add(propertyNameWithUnderscores.toString());
+
+        return columnName;
+    }
+
+    /**
+     * Gets the locale used to manipulate the Bean properties name. This locale is useful, for example, when doing
+     * case conversion. The locale is expected to be constant across method invocations. By default the {@code US}
+     * locale will be used.
+     *
+     * @return the target locale, never {@code null}
+     */
+    protected Locale getLocale() {
+        return Locale.US;
```

Comment:
Do you have some examples to justify making US locale the default for everyone worldwide?

This looks a bit heavy-handed.